### PR TITLE
Add metadata to install-config and provide CIS CRN through it

### PR DESCRIPTION
**Pull request by mjturek**

### CODE REVIEW

**Comment by mjturek** [comment]
Okay this patch seems to work as long as you export IC_API_KEY

**Comment by clnperez** [comment]
Ah, so it looks like, if users don't have `IC_API_KEY` set, this doesn't work: https://github.com/openshift-powervs/installer/blob/add-ipi/pkg/asset/installconfig/ibmcloud/client.go#L82 

I hadn't thought we'd have to require users to set this. Also we'd planned on allowing users to set other vars that have been supported in the past, or having their key stored on disk.  (fyi @mkumatag )

Is this what @23TNC was talking about having to export to an env var if it wasn't set?

**Comment by mjturek** [comment]
@clnperez yep, this is the issue Cory was talking about. The way @23TNC is handling this is setting `IC_API_KEY` using `os.setenv()`. We can do that here as well to ensure that it's set properly. Correct me if I'm wrong @mkumatag but last call you agreed this solution was sufficient

**Comment by clnperez** [comment]
Can you also add this CRN value in to the Metadata in pkg/asset/cluster/powervs/powervs.go and set it there?

**Comment by mjturek** [comment]
> Can you also add this CRN value in to the Metadata in pkg/asset/cluster/powervs/powervs.go and set it there?

Good catch! Done!

File: pkg/asset/installconfig/powervs/metadata.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/openshift/installer/pkg/asset/installconfig/ibmcloud"
+)
+
+// Metadata holds additional metadata for InstallConfig resources that
+// does not need to be user-supplied (e.g. because it can be retrieved
```
Comment (clnperez) [inline pkg/asset/installconfig/powervs/metadata.go]: nit: does not need => do not ned

**Review by clnperez — CHANGES_REQUESTED** [review]
LGTM, except for a grammar nit

File: pkg/asset/installconfig/powervs/metadata.go (hunk 1 above)
Comment (mjturek) [reply pkg/asset/installconfig/powervs/metadata.go]: Came from a copy paste but can fix :)

**Pull request merged**